src/Apcu/Apcu.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Apcu;
+
+/**
+ * @internal
+ */
+final class Apcu
+{
+    public static function apcu_entry($key, $generator, $ttl = 0)
+    {
+        if (!is_scalar($key)) {
+            throw new \InvalidArgumentException(sprintf('apcu_entry() expects parameter 1 to be string, %s given', __FUNCTION__, gettype($value)));
+        }
+
+        $entry = apcu_fetch($key, $success);
+
+        if ($success) {
+            return $entry;
+        }
+
+        $h = fopen(__FILE__, 'r');
+        flock($h, LOCK_EX);
```

Comment:
Is it sufficient to avoid race conditions?
What if the file is already being executed by another thread before the call to `flock`? What's the behavior when the opcache caching is on?

Maybe can you just throw a `NotSupportedException` for this specific function?